mailing_list/models.py

```diff
@@ -90,7 +133,9 @@ def get_or_create_or_delete_mailing_lists_for_canvas_course_id(self, canvas_cour
 
         # Delete existing mailing lists who's section no longer exists
         for section_id, mailing_list in mailing_lists_by_section_id.iteritems():
-            mailing_list.delete()
+            # we don't want to delete the course list here
+            if section_id:
```

Comment:
You won't need this anymore if you pop the course list as described in the change above.